Major/lifecycle improvements

jlacivita opened the pull request.

jlacivita committed:
feat: Added Manage APIs for `windowColor` and `windowOpacity` (#164)

# [0.15.0-next.5](https://github.com/rdkcentral/firebolt-apis/compare/v0.15.0-next.4...v0.15.0-next.5) (2023-07-31)

### Bug Fixes

* Upgrade to `firebolt-js/openrpc` version 2.0.3 ([c0a2647](https://github.com/rdkcentral/firebolt-apis/commit/c0a26475dba5059f9fc6ceb11bcb103bb355cceb))

* feat: Added manage apis for windlowColor and windowOpacity (#157)

* feat: Added manage apis for windlowColor and windowOpacity

* chore(release): 0.16.0-next.1 [skip ci]

# [0.16.0-next.1](https://github.com/rdkcentral/firebolt-apis/compare/v0.15.0...v0.16.0-next.1) (2023-08-07)

### Features

* Added manage apis for windlowColor and windowOpacity ([#157](https://github.com/rdkcentral/firebolt-apis/issues/157)) ([f508358](https://github.com/rdkcentral/firebolt-apis/commit/f508358aeb2f58bb3893bbfaf09e2340fcadac8f))

* fix: Unit test cases for setting null value (#159)

fix: Unit test cases for setting null value

* fix: Update firebolt-openrpc, fix tests

* chore(release): 0.16.0-next.2 [skip ci]

# [0.16.0-next.2](https://github.com/rdkcentral/firebolt-apis/compare/v0.16.0-next.1...v0.16.0-next.2) (2023-08-10)

### Bug Fixes

* Unit test cases for setting null value ([#159](https://github.com/rdkcentral/firebolt-apis/issues/159)) ([7e011e2](https://github.com/rdkcentral/firebolt-apis/commit/7e011e2937842902c6b111646492b33a057b23c8))
* Update firebolt-openrpc, fix tests ([9480984](https://github.com/rdkcentral/firebolt-apis/commit/948098490595ffa374df08f3f44157d70beb7990))
* Update firebolt-openrpc, fix tests ([312b5ec](https://github.com/rdkcentral/firebolt-apis/commit/312b5ec627e9b494ac455475efe7851bed70c34e))

* chore: Point to product `firebolt-openrpc` release

* chore(release): 0.16.0-next.3 [skip ci]

# [0.16.0-next.3](https://github.com/rdkcentral/firebolt-apis/compare/v0.16.0-next.2...v0.16.0-next.3) (2023-08-14)

### Bug Fixes

* Update package.json to point to release 2.0.4 of `firebolt-openrpc` ([9e3cf55](https://github.com/rdkcentral/firebolt-apis/commit/9e3cf55d465d3b429ebcba0167e888eb7d7f0588))

* Update CHANGELOG.md

---------

Co-authored-by: semantic-release-bot <semantic-release-bot@martynus.net>
Co-authored-by: Kevin <72702438+kevinshahfws@users.noreply.github.com>

semantic-release-bot committed:
chore(release): 0.16.0 [skip ci]

# [0.16.0](https://github.com/rdkcentral/firebolt-apis/compare/v0.15.0...v0.16.0) (2023-08-14)

### Features

* Added Manage APIs for `windowColor` and `windowOpacity` ([#164](https://github.com/rdkcentral/firebolt-apis/issues/164)) ([99b9fa6](https://github.com/rdkcentral/firebolt-apis/commit/99b9fa60d3987e7a8d5e7230e7401cae7d2d352c)), closes [#157](https://github.com/rdkcentral/firebolt-apis/issues/157) [#157](https://github.com/rdkcentral/firebolt-apis/issues/157) [#159](https://github.com/rdkcentral/firebolt-apis/issues/159) [#159](https://github.com/rdkcentral/firebolt-apis/issues/159)

jlacivita committed:
fix: Reduce font size on capability index headers

jlacivita committed:
fix: Using 3 letter ISO639 language codes (#173)

* fix: Using 3 letter ISO639 language codes

* fix: Various PR comments resolved

* fix: Typo and schemas path priority

semantic-release-bot committed:
chore(release): 0.16.0-next.4 [skip ci]

# [0.16.0-next.4](https://github.com/rdkcentral/firebolt-apis/compare/v0.16.0-next.3...v0.16.0-next.4) (2023-09-01)

### Bug Fixes

* Reduce font size on capability index headers ([6c1487b](https://github.com/rdkcentral/firebolt-apis/commit/6c1487bf65ac005028c60759bcace8550156e987))
* **transport:** Add test for lazy-loading the transport layer ([#10](https://github.com/rdkcentral/firebolt-apis/issues/10)) ([168fcaf](https://github.com/rdkcentral/firebolt-apis/commit/168fcaf4e6264cc0c2498f02c16523fb4a51cdd3)), closes [#164](https://github.com/rdkcentral/firebolt-apis/issues/164) [#157](https://github.com/rdkcentral/firebolt-apis/issues/157) [#157](https://github.com/rdkcentral/firebolt-apis/issues/157) [#159](https://github.com/rdkcentral/firebolt-apis/issues/159) [#159](https://github.com/rdkcentral/firebolt-apis/issues/159)
* Using 3 letter ISO639 language codes ([#173](https://github.com/rdkcentral/firebolt-apis/issues/173)) ([b341dee](https://github.com/rdkcentral/firebolt-apis/commit/b341deedd9ac8ab00f3e762622fda29872f83932))

jlacivita committed:
Merge branch 'main' into next

jlacivita committed:
Merge branch 'next' of github.com:rdkcentral/firebolt-apis into next

jlacivita committed:
fix: Manual merge of main to next

semantic-release-bot committed:
chore(release): 0.17.0-next.1 [skip ci]

# [0.17.0-next.1](https://github.com/rdkcentral/firebolt-apis/compare/v0.16.0...v0.17.0-next.1) (2023-09-01)

### Bug Fixes

* Manual merge of main to next ([6a11710](https://github.com/rdkcentral/firebolt-apis/commit/6a11710d3c8028ebbe70b39a27f4944184b93181))
* Reduce font size on capability index headers ([6c1487b](https://github.com/rdkcentral/firebolt-apis/commit/6c1487bf65ac005028c60759bcace8550156e987))
* **transport:** Add test for lazy-loading the transport layer ([#10](https://github.com/rdkcentral/firebolt-apis/issues/10)) ([168fcaf](https://github.com/rdkcentral/firebolt-apis/commit/168fcaf4e6264cc0c2498f02c16523fb4a51cdd3)), closes [#164](https://github.com/rdkcentral/firebolt-apis/issues/164) [#157](https://github.com/rdkcentral/firebolt-apis/issues/157) [#157](https://github.com/rdkcentral/firebolt-apis/issues/157) [#159](https://github.com/rdkcentral/firebolt-apis/issues/159) [#159](https://github.com/rdkcentral/firebolt-apis/issues/159)
* Unit test cases for setting null value ([#159](https://github.com/rdkcentral/firebolt-apis/issues/159)) ([7e011e2](https://github.com/rdkcentral/firebolt-apis/commit/7e011e2937842902c6b111646492b33a057b23c8))
* Update firebolt-openrpc, fix tests ([9480984](https://github.com/rdkcentral/firebolt-apis/commit/948098490595ffa374df08f3f44157d70beb7990))
* Update firebolt-openrpc, fix tests ([312b5ec](https://github.com/rdkcentral/firebolt-apis/commit/312b5ec627e9b494ac455475efe7851bed70c34e))
* Update package.json to point to release 2.0.4 of `firebolt-openrpc` ([9e3cf55](https://github.com/rdkcentral/firebolt-apis/commit/9e3cf55d465d3b429ebcba0167e888eb7d7f0588))
* Using 3 letter ISO639 language codes ([#173](https://github.com/rdkcentral/firebolt-apis/issues/173)) ([b341dee](https://github.com/rdkcentral/firebolt-apis/commit/b341deedd9ac8ab00f3e762622fda29872f83932))

### Features

* Added manage apis for windlowColor and windowOpacity ([#157](https://github.com/rdkcentral/firebolt-apis/issues/157)) ([f508358](https://github.com/rdkcentral/firebolt-apis/commit/f508358aeb2f58bb3893bbfaf09e2340fcadac8f))

jlacivita committed:
fix: update to firebolt-openrpc, c tests (#176)

* native tests added (#140)

* native tests added

* review updates

* Test app alignment based on change in the static/generated code

* ctest: add WPEFrameworkCore also in the dependency

* Native test update based on interface return type change from uint32_t to int32_t

* fix: update to npm build of openrpc

* fix: Update packaeg-lock.json

---------

Co-authored-by: HaseenaSainul <41037131+HaseenaSainul@users.noreply.github.com>

semantic-release-bot committed:
chore(release): 0.17.0-next.2 [skip ci]

# [0.17.0-next.2](https://github.com/rdkcentral/firebolt-apis/compare/v0.17.0-next.1...v0.17.0-next.2) (2023-09-05)

### Bug Fixes

* update to firebolt-openrpc, c tests ([#176](https://github.com/rdkcentral/firebolt-apis/issues/176)) ([3984a78](https://github.com/rdkcentral/firebolt-apis/commit/3984a781d871ab9b35aecbab77157b635e4f638f)), closes [#140](https://github.com/rdkcentral/firebolt-apis/issues/140)

jlacivita committed:
fix: Point to production firebolt-openrpc

semantic-release-bot committed:
chore(release): 0.17.0-next.3 [skip ci]

# [0.17.0-next.3](https://github.com/rdkcentral/firebolt-apis/compare/v0.17.0-next.2...v0.17.0-next.3) (2023-09-07)

### Bug Fixes

* Point to production firebolt-openrpc ([a367307](https://github.com/rdkcentral/firebolt-apis/commit/a367307d45dbdebfd3d1bf17a87113bc220ea93f))

jlacivita committed:
Merge pull request #178 from rdkcentral/next

Release 0.17.0

jlacivita committed:
Update CHANGELOG.md

jlacivita committed:
Update CHANGELOG.md

semantic-release-bot committed:
chore(release): 0.17.0 [skip ci]

# [0.17.0](https://github.com/rdkcentral/firebolt-apis/compare/v0.16.0...v0.17.0) (2023-09-07)

### Bug Fixes

* Manual merge of main to next ([6a11710](https://github.com/rdkcentral/firebolt-apis/commit/6a11710d3c8028ebbe70b39a27f4944184b93181))
* Point to production firebolt-openrpc ([a367307](https://github.com/rdkcentral/firebolt-apis/commit/a367307d45dbdebfd3d1bf17a87113bc220ea93f))
* Reduce font size on capability index headers ([6c1487b](https://github.com/rdkcentral/firebolt-apis/commit/6c1487bf65ac005028c60759bcace8550156e987))
* **transport:** Add test for lazy-loading the transport layer ([#10](https://github.com/rdkcentral/firebolt-apis/issues/10)) ([168fcaf](https://github.com/rdkcentral/firebolt-apis/commit/168fcaf4e6264cc0c2498f02c16523fb4a51cdd3)), closes [#164](https://github.com/rdkcentral/firebolt-apis/issues/164) [#157](https://github.com/rdkcentral/firebolt-apis/issues/157) [#157](https://github.com/rdkcentral/firebolt-apis/issues/157) [#159](https://github.com/rdkcentral/firebolt-apis/issues/159) [#159](https://github.com/rdkcentral/firebolt-apis/issues/159)
* Unit test cases for setting null value ([#159](https://github.com/rdkcentral/firebolt-apis/issues/159)) ([7e011e2](https://github.com/rdkcentral/firebolt-apis/commit/7e011e2937842902c6b111646492b33a057b23c8))
* Update firebolt-openrpc, fix tests ([9480984](https://github.com/rdkcentral/firebolt-apis/commit/948098490595ffa374df08f3f44157d70beb7990))
* Update firebolt-openrpc, fix tests ([312b5ec](https://github.com/rdkcentral/firebolt-apis/commit/312b5ec627e9b494ac455475efe7851bed70c34e))
* Update package.json to point to release 2.0.4 of `firebolt-openrpc` ([9e3cf55](https://github.com/rdkcentral/firebolt-apis/commit/9e3cf55d465d3b429ebcba0167e888eb7d7f0588))
* update to firebolt-openrpc, c tests ([#176](https://github.com/rdkcentral/firebolt-apis/issues/176)) ([3984a78](https://github.com/rdkcentral/firebolt-apis/commit/3984a781d871ab9b35aecbab77157b635e4f638f)), closes [#140](https://github.com/rdkcentral/firebolt-apis/issues/140)
* Using 3 letter ISO639 language codes ([#173](https://github.com/rdkcentral/firebolt-apis/issues/173)) ([b341dee](https://github.com/rdkcentral/firebolt-apis/commit/b341deedd9ac8ab00f3e762622fda29872f83932))

### Features

* Added manage apis for windlowColor and windowOpacity ([#157](https://github.com/rdkcentral/firebolt-apis/issues/157)) ([f508358](https://github.com/rdkcentral/firebolt-apis/commit/f508358aeb2f58bb3893bbfaf09e2340fcadac8f))

jlacivita committed:
fix: Deploy GitHub.io docs for PRs (#180)

* chore: Set up PR CI

* Update README.md

* Update release-github.yml

* fix: testing ci for prs

* fix: build artifacts when github action is a PR

* fix: Adding debug

* fix: Fixed dangling quote

* Revert "fix: testing ci for prs"

This reverts commit 73edcef93f71325b079c8dc1898b5cc3b007e707.

* Revert "Update README.md"

This reverts commit b85411733af954d3193c179c80f95b878f282245.

* fix: Remove test branches

* fix: silently skip git commit if there's no docs changes

* fix: Adding proposed branch to CI

* chore: Set up automatic back-merge

* fix: Update semantic-release version

semantic-release-bot committed:
chore(release): 0.17.1-next.1 [skip ci]

## [0.17.1-next.1](https://github.com/rdkcentral/firebolt-apis/compare/v0.17.0...v0.17.1-next.1) (2023-09-08)

### Bug Fixes

* Deploy GitHub.io docs for PRs ([#180](https://github.com/rdkcentral/firebolt-apis/issues/180)) ([7ef7d9d](https://github.com/rdkcentral/firebolt-apis/commit/7ef7d9dac20d73591b6c3391a8afbbfd55a717e7))

jlacivita committed:
fix: Add proposed branch to CI builds

semantic-release-bot committed:
chore(release): 0.17.1-next.2 [skip ci]

## [0.17.1-next.2](https://github.com/rdkcentral/firebolt-apis/compare/v0.17.1-next.1...v0.17.1-next.2) (2023-09-12)

### Bug Fixes

* Add proposed branch to CI builds ([8a5c142](https://github.com/rdkcentral/firebolt-apis/commit/8a5c142def4b50a2f83836bf85b8c51889793890))

jlacivita committed:
fix: Add proposed branch to .releaserc

jlacivita committed:
Merge branch 'next' of github.com:rdkcentral/firebolt-apis into next

semantic-release-bot committed:
chore(release): 0.17.1-next.3 [skip ci]

## [0.17.1-next.3](https://github.com/rdkcentral/firebolt-apis/compare/v0.17.1-next.2...v0.17.1-next.3) (2023-09-12)

### Bug Fixes

* Add proposed branch to .releaserc ([e65e43f](https://github.com/rdkcentral/firebolt-apis/commit/e65e43f47113e72b79c09d9fa89b2d9d3dad9e9d))

jlacivita committed:
fix: Add enumerated values for fontFamily/fontEdge (#181)

* fix: Add enumerated values for fontFamily/fontEdge
* fix: Fixing example to match new enums

semantic-release-bot committed:
chore(release): 0.17.1-next.4 [skip ci]

## [0.17.1-next.4](https://github.com/rdkcentral/firebolt-apis/compare/v0.17.1-next.3...v0.17.1-next.4) (2023-09-12)

### Bug Fixes

* Add enumerated values for fontFamily/fontEdge ([#181](https://github.com/rdkcentral/firebolt-apis/issues/181)) ([de35cb8](https://github.com/rdkcentral/firebolt-apis/commit/de35cb8073438f945bb4b7a8e8f420b8d6ba2358))

jlacivita committed:
Merge pull request #182 from rdkcentral/next

Release 0.17.1

jlacivita committed:
Update CHANGELOG.md

jlacivita committed:
Update CHANGELOG.md

semantic-release-bot committed:
chore(release): 0.17.1 [skip ci]

## [0.17.1](https://github.com/rdkcentral/firebolt-apis/compare/v0.17.0...v0.17.1) (2023-09-15)

### Bug Fixes

* Add enumerated values for fontFamily/fontEdge ([#181](https://github.com/rdkcentral/firebolt-apis/issues/181)) ([de35cb8](https://github.com/rdkcentral/firebolt-apis/commit/de35cb8073438f945bb4b7a8e8f420b8d6ba2358))
* Add proposed branch to .releaserc ([e65e43f](https://github.com/rdkcentral/firebolt-apis/commit/e65e43f47113e72b79c09d9fa89b2d9d3dad9e9d))
* Add proposed branch to CI builds ([8a5c142](https://github.com/rdkcentral/firebolt-apis/commit/8a5c142def4b50a2f83836bf85b8c51889793890))
* Deploy GitHub.io docs for PRs ([#180](https://github.com/rdkcentral/firebolt-apis/issues/180)) ([7ef7d9d](https://github.com/rdkcentral/firebolt-apis/commit/7ef7d9dac20d73591b6c3391a8afbbfd55a717e7))

kevinshahfws committed:
feat: Added optional parameter 'force' in usergrants.request (#186)

* feat: Added optional parameter 'force' in usergrants.request
* Addressed review comments

semantic-release-bot committed:
chore(release): 0.18.0-next.1 [skip ci]

# [0.18.0-next.1](https://github.com/rdkcentral/firebolt-apis/compare/v0.17.1...v0.18.0-next.1) (2023-09-26)

### Features

* Added optional parameter 'force' in usergrants.request ([#186](https://github.com/rdkcentral/firebolt-apis/issues/186)) ([31801ca](https://github.com/rdkcentral/firebolt-apis/commit/31801caec6bea0e8b295ea6a9ec54ca1d8e08d16))

kevinshahfws committed:
fix: Updated metrics:distributor capability to could (#185)

semantic-release-bot committed:
chore(release): 0.18.0-next.2 [skip ci]

# [0.18.0-next.2](https://github.com/rdkcentral/firebolt-apis/compare/v0.18.0-next.1...v0.18.0-next.2) (2023-09-26)

### Bug Fixes

* Updated metrics:distributor capability to could ([#185](https://github.com/rdkcentral/firebolt-apis/issues/185)) ([9ca2206](https://github.com/rdkcentral/firebolt-apis/commit/9ca2206a2dca7149dcf93df3d303806d136785ed))

jlacivita committed:
feat: First Draft based on today's working group

jlacivita committed:
fix: Adding more details

jlacivita committed:
fix: Add some context for events vs transitions

jlacivita committed:
fix: more ntoes

jlacivita committed:
Update index.md

jlacivita committed:
fix: Aligning Presentation w/ Lifecycle

jlacivita committed:
fix: Updates from today's WG

jlacivita committed:
fix: Added code example

jlacivita committed:
fix: Typos

jlacivita committed:
chore: Added todos from last meeting

kevinshahfws committed:
fix: Updated Discovery.launch example with xrn:firebolt:application-type (#187)

* fix: Updated Discovery.launch example with xrn:firebolt:application-type
* fix: addressed review comments

kevinshahfws committed:
feature: Added scope parameter for advertisingId (#188)

* feature: Added scope.type and scope.id option parameters for advertisingId api
* Addressed review comments

semantic-release-bot committed:
chore(release): 0.18.0-next.3 [skip ci]

# [0.18.0-next.3](https://github.com/rdkcentral/firebolt-apis/compare/v0.18.0-next.2...v0.18.0-next.3) (2023-10-10)

### Bug Fixes

* Updated Discovery.launch example with xrn:firebolt:application-type ([#187](https://github.com/rdkcentral/firebolt-apis/issues/187)) ([4dbbab3](https://github.com/rdkcentral/firebolt-apis/commit/4dbbab3d9fa68c0e5185ca72fd0170bae8a30139))

jlacivita committed:
fix: Latest working group changes

jlacivita committed:
Cleaned up spec based on working group consensus

jlacivita committed:
Update index.md

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
-the initializing state. Apps **MUST** remain in this state until they
-register with the platform via Lifecycle.ready(), or
-Lifecycle.onInactive(), or are unloaded by the platform.
+When an app starts running is **MUST** initialize the Firebolt SDK as quickly as possible
```
small typo here is -> it

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
-register with the platform via Lifecycle.ready(), or
-Lifecycle.onInactive(), or are unloaded by the platform.
+When an app starts running is **MUST** initialize the Firebolt SDK as quickly as possible
+and then wait for the `Application.create()` inteface to be called before doing further setup.
```
inteface -> interface

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
+
+Apps in this state **MUST** have access to the graphics compositor.
 
 Apps in this state **SHOULD NOT** have performance negatively impacted
```
I know this one is carry over, but not sure we can say this will always be true?

DeltaHeadMan commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
 
 | Contributor    | Organization   |
 | -------------- | -------------- |
 | Andrew Bennet            | Sky            |
```
Two t's in Bennett.

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
+Apps in this state **MUST NOT** have access to the audio-video decoder.
 
-### 2.5. Unloading
+Apps in this state **SHOULD NOT** have performance negatively impacted
```
Don't think this belongs here? Apps in suspended will have less access to CPU and thus should have performance negatively impacted?

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
+capability, then the following requirements **MUST** be fulfilled.
 
-Apps **MUST** only enter this state from the inactive state.
+Apps **MUST** only enter this state from the `RUNNING` state, via
```
Maybe this is copy/paste issue? This is describing the suspended state, not sleeping.
We probably do need this in the Suspended section too though

DeltaHeadMan commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
 app.
 
-*Unloading* an app refers to the process of notifying an app to do any
+*Suspending* an app refers to reducing the app's memory usage and CPU
```
"memory and CPU usage"

DeltaHeadMan commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
+*Resuming* an app refers to reallocating its graphics compositors and
+resuming normal CPU and memory allocations.
+
+*Destroying* / unloading an app refers to the process of notifying an app to do any
```
I think it's just "destroying", not "destroying / unloading"

jlacivita replied:
Okay, let's change all "loading" to "initializing and all "unloading" to"destroying".

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
 
 *Immediately before* an app transitions to this state, the platform
-**MUST** dispatch the Lifecycle.onSuspended notification with the
+**MUST** dispatch the Lifecycle.onSleeping notification with the
```
This one is interesting, not sure we can dispatch the state change because the app will have no CPU. And any code that listens for this probably won't be able to execute.

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
-for any reason, then the app **MUST** be treated as though it were
-unloaded.
+The platform **MUST** save the app's memory space at this point, so
+that it may be unsuspended later. If storing the app\'s memory space
```
unsuspended probably wrong term here, probably "that it may be woken later"

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
+The platform **MUST** save the app's memory space at this point, so
+that it may be unsuspended later. If storing the app\'s memory space
+fails for any reason, then the app **MUST** be treated as though it
+were unloaded.
```
unloaded or terminated?

DeltaHeadMan commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
 *Terminating* an app refers to shutting down the app's container and
 reclaiming any memory, CPU, etc. that it was consuming.
 
+*Sleeping* an app refers to turning off all CPU cycles for that app
```
"refers to putting the app into a state where it does not use CPU cycles, which allows the platform to optimise memory consumption.

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
+fails for any reason, then the app **MUST** be treated as though it
+were unloaded.
+
+Finally, the app and its container **MUST** be removed from memory
```
This one may be more nuanced. I think some memory may remain even in sleep.

kpears201 replied:
Some sleep implementations also might do RAM compression instead of writing to disk.

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -287,189 +378,301 @@ manifest's releaseStatus is not developer.
-the platform **MUST** transition the app into the background state. This
-enables apps to display a custom loading screen on devices that support
-it. Supporting this capability is optional.
+Initialzing consists of three parts:
```
Initialzing -> Initializing

DeltaHeadMan commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
+
+### 4.1. Loading an app
+
+**TODO**: Should this be it's own spec, and out of scope?
```
not in scope of this spec, I think.

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -287,189 +378,301 @@ manifest's releaseStatus is not developer.
+If the app is already in the `ACTIVE` state, then it is already
+activated and there is no need to do anything else. The platform **MUST
+NOT** dispatch any *additional* lifecycle notifications when attempting
+to activate an app that is already in the foreground state and the
```
foreground?

DeltaHeadMan commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -287,189 +378,301 @@ manifest's releaseStatus is not developer.
+> and is ready to be activated and presented to the user.
+>
+> If the app times out or makes a `createError` call, then the app
+> **MUST** be terminated.
```
immediately terminated by the platform.

DeltaHeadMan commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -287,189 +378,301 @@ manifest's releaseStatus is not developer.
 
-If the app is suspended, then it **MUST** be *unsuspended* first.
+Once the `create` method returns the app **MUST** be transitioned to
+the `RUNNING` state within 100 milliseconds.
```
or just say "be immediately transitioned".

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -287,189 +378,301 @@ manifest's releaseStatus is not developer.
+>
+> The `onRequestActivate` call **MUST** include a `NavigationIntent`.
+>
+> If the app sends a `Lifecycle.activateFocus` request and has
```
Not sure this should be called focus. I don't think the app should have key focus when the loading screen is up.

jlacivita replied:
Let's discuss when we talk about loading screens tomorrow

DeltaHeadMan commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -287,189 +378,301 @@ manifest's releaseStatus is not developer.
-Unsuspending an app loads it's previous state back into memory and
-transitions it to the inactive state, so that it may be launched back
-into its original state.
+**TODO**: discuss ^^
```
... "reduce memory usage, if possible".

DeltaHeadMan commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -479,81 +682,255 @@ scope for this document.
-ResourceDeallocator implementation.
+If an app provides the `xrn:firebolt:capability:lifecycle:application`
+capability, then the platform **MUST** call the app's implementation of
+`Application.unload()`:
```
isn't this destroy() ?

DeltaHeadMan commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -479,81 +682,255 @@ scope for this document.
+
+The `status` API must have one of the following values:
+
+- `Initializing`
```
would it be easier to show the current state, and the new state instead?

kpears201 replied:
think I agree with above, instead of inventing new language for the transition just use the existing language for the start and end of the transition.

DeltaHeadMan commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -287,189 +378,301 @@ manifest's releaseStatus is not developer.
-**MUST NOT** dispatch any *additional* lifecycle notifications when
-attempting to background an app that is already in the background state
-and the remainder of this section does not need to happen.
+If an app provides the `xrn:firebolt:capability:lifecycle:application`
```
isn't this: xrn:...:activity ?

kpears201 replied:
application -> activity ?

DeltaHeadMan commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -287,189 +378,301 @@ manifest's releaseStatus is not developer.
-Apps that are in the foreground state **MAY** request to be
-backgrounded, via the Lifecycle.background() API method. See the
-Background API below.
+If an app is currently [unloading](#38-unloading-an-app) then the app 
```
unloading has been removed.

DeltaHeadMan commented on requirements/specifications/lifecycle/presentation.md:
```diff
@@ -0,0 +1,121 @@
+
+| Contributor    | Organization   |
+| -------------- | -------------- |
+| Andrew Bennet            | Sky            |
```
Two t's.

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -287,189 +378,301 @@ manifest's releaseStatus is not developer.
-At this point, the app **MUST** be in the inactive state.
+At this point, the app **MUST** be in the `RUNNING` state.
+
+If an app provides the `xrn:firebolt:capability:lifecycle:application`
```
Do we need this as an "If"? The app MUST provide this right?

jlacivita replied:
Remove all of these "Ifs"

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -287,189 +378,301 @@ manifest's releaseStatus is not developer.
+Resuming an app allows it to reallocate graphics composition and
+reload any resources it might have deallocated during `suspend()`.
+
+The platform may unsuspend apps so they can access more resources or
```
unsuspend -> resume?

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -287,189 +378,301 @@ manifest's releaseStatus is not developer.
 To unsuspend an app, platforms **MUST** use the following process.
 
-If an app is not in the suspended state, then it cannot be unsuspended
+If an app is not in the `SUSPENDED` state, then it cannot be unsuspended
```
unsuspended -> resumed?

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -287,189 +378,301 @@ manifest's releaseStatus is not developer.
+
+Firebolt apps that have permission to use the `xrn:firebolt:capability:lifecycle:sleep` capability **MUST** implement `Sleepable.wake()`.
+
+### 4.9. Unloading an app
```
Andrew has pointed this out in a number of places but sounds like we should remove all mentions of "unloading" in favor of destroying.

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -479,81 +682,255 @@ scope for this document.
+ - Performing any entitlement checks to decide whether to display a player or a purchase flow
+ - Any other steps necesary to present content to the user quickly
+
+## 7. Launch Parameters
```
I think we should avoid the term "Launch" since it is ambiguous with the states.
Maybe this should be called `Create Parameters`?

jlacivita replied:
Discuss and agreed to keep  as is

DeltaHeadMan commented on requirements/specifications/lifecycle/presentation.md:
```diff
@@ -0,0 +1,121 @@
+As a property, this API also has an `onFocusChanged` notification.
+
+## 3. Display
+The `Presentation` module **MUST** have a `display` string property that returns one of the following values:
```
is this an enumeration?

bcatrysse commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
+Running apps can execute code, but are not [Presented](./presentation.md) to the user, do not receive
+any input from RCUs, and cannot use the video decoder.
+
+Apps in this state **MUST** have access to the graphics compositor.
```
Apps in this state **MAY or SHOULD** have created, allocated a Graphics Surface but it will not be made visible by platform compositor.

bcatrysse commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
+once.
 
-Apps in this state **MUST** be visible.
+Apps in this state **MAY** be visible.
```
Depending on Display value in [Presentation](./presentation.md) the App will be visible in this state or not

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
 
-Platforms **MAY** support the app lifecycle states and transitions
-denoted in gray.
+These transition APIs are asynchronous, and each one has a platform-configurable timeout that specifies how long the app has to fulfill the method.
```
Yeah, Firebolt is waiting until the APIs return,so I think that makes it sync?
But maybe we just don't mention async/sync here since it could be confusing?

kpears201 commented on requirements/specifications/lifecycle/presentation.md:
```diff
@@ -0,0 +1,121 @@
+When an app has this capability, it **MAY** be put into the `none` display state while in the `active` Lifecycle state.
+
+## 6. Picture-in-Picture Video
+If an app has the `xrn:firebolt:capability:media:picture-in-picture`, then it can keep playing audio/video when the app is in the `none` display state and the audio & video will be presented to the user in bounding box determined by the platform. Note that this does not include the entire UX of the app, just the active media pipeline.
```
Do we need to inform the app that this is happening?

kpears201 commented on requirements/specifications/lifecycle/presentation.md:
```diff
@@ -0,0 +1,121 @@
+
+If an app provides the `xrn:firebolt:capability:presentation:loading-screen`
+capability, then the platform **MAY** use the app-provided loading screen, in
+which case, the rest of the section does not apply.
```
I think the rest still applies in that case since it may take time to load the loading screen?

kpears201 commented on requirements/specifications/lifecycle/presentation.md:
```diff
@@ -0,0 +1,121 @@
+which case, the rest of the section does not apply.
+
+The loading screen **SHOULD** include a loading image referenced in the app's
+manifest and cached on the device.
```
cached on the device is maybe an impl detail?

kpears201 replied:
also maybe we should avoid terms like manifest since that hasn't been spec'd

bcatrysse commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
-resources. See [Background App
-Requirements](https://comcastcorp.sharepoint.com/:w:/r/sites/hqppa/Shared%20Documents/Global%20App%20Platform/Firebolt/Requirements/Core/Lifecycle/Background%20App%20Requirements.docx?d=w185d9112bcd44376bb0635dad816919b&csf=1&web=1&e=cjmuSJ)
-for more info.
+Apps in this state **MUST NOT** have access to the graphics compositor.
```
Apps in this state **MUST NOT** have associated Graphics Surface created/allocated. Need to destroy, deallocated that before moving to this state as part of suspend() call

bcatrysse commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
+|                | CPU | RAM | Net | GFX | A/V | Description                                                                   |
+|----------------|-----|-----|-----|-----|-----|-------------------------------------------------------------------------------|
+| `create()`     | ✔   | ✔   | ✔   | ✔   |     | Creation of the app, from a Firebolt Lifecycle perspective.                   |
+| `suspend()`    | ✔   | ✔   | ✔   | ✔   |     | Reduce memory usage and CPU cycles, deallocate graphics compositors.          |
```
Deallocate associated Graphics Surface

kpears201 commented on requirements/specifications/lifecycle/presentation.md:
```diff
@@ -0,0 +1,121 @@
+but invoked for the app. 
+
+Apps that provide the loading screen capability **MUST** be made visible at
+the very beginning of the `Activity.activate()` transition, rather
```
Don't we need some lines on app requirements of when they render their loading screen?
If we are going to make the app visible at the beginning of activate, I think that means the app MUST have the loading screen rendered at the end of create.

bcatrysse commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -1,274 +1,365 @@
 
-*Unloading* an app refers to the process of notifying an app to do any
+*Suspending* an app refers to reducing the app's memory usage and CPU
+cycles, and deallocating its graphics compositors so that other apps
```
deallocating its associated graphics surface

bcatrysse reviewed (changes requested):
use graphics surface instead of graphics compositors

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -479,81 +682,255 @@ scope for this document.
-### 4.1. Ready
+### 8.5. Ready
 
 The Lifecycle.ready() API allows an app to notify the platform that it
```
Remove this section now?

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -479,81 +682,255 @@ scope for this document.
 
 Platforms generally **SHOULD** respect this call and move the app to the
-inactive state, but there may be edge cases where this is not possible,
+started state, but there may be edge cases where this is not possible,
```
"running" state

kpears201 commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -565,99 +942,15 @@ provided. For example, apps closed due to the RCU are less likely to be
+## 9. Lifecycle Configuration
 
-## 6. Lifecycle Configuration
+**TODO**: do we want these to be per spec, per distributor, or per app?
```
per distributor per device

jlacivita commented on requirements/specifications/lifecycle/index.md:
```diff
@@ -667,9 +960,13 @@ for state transitions.
-| appFinishedTimeout | bigint | Yes      | Number of milliseconds the platform should wait after dispatching the onUnloading notification before finally unloading an app that did not call finished().                                     |
-| appSuspendTimeout  | bigint | Yes      | Number of milliseconds the platform should wait after dispatching the onRequestSuspend notification before finally suspending (or terminating) an app that did not call onSuspendResponse().     |
-| appInactiveTimeout | bigint | Yes      | Number of milliseconds the platform should wait after dispatching the onRequestInactive notification before finally inactivating (or terminating) an app that did not call onInactiveResponse(). |
+| appCreateTimeout    | bigint | Yes      | Number of milliseconds the platform should wait before terminating a loaded app that did not finish `create()`.                                                                              |
```
yup, will add.

jlacivita committed:
fix: Updates from today's working group

jlacivita committed:
fix: table clean up

jlacivita committed:
chore: Added Markdown Beautifier and link fixer

jlacivita committed:
fix: Today's working group edits

jlacivita committed:
fix: Presentation Working Group Edits

jlacivita committed:
chore: minor formatting

kevinshahfws committed:
fix: Account.session params to be mandatory (#196)

* fix: Updated params in Account.session to mandatory

* fix: Updated storage secure capability to could

semantic-release-bot committed:
chore(release): 0.18.0-next.4 [skip ci]

# [0.18.0-next.4](https://github.com/rdkcentral/firebolt-apis/compare/v0.18.0-next.3...v0.18.0-next.4) (2023-10-24)

### Bug Fixes

* Account.session params to be mandatory ([#196](https://github.com/rdkcentral/firebolt-apis/issues/196)) ([fc5c638](https://github.com/rdkcentral/firebolt-apis/commit/fc5c63886d9b4eb30b32c1edc75f0f6afe80a827))

jlacivita committed:
chore: whitespace cleanup

jlacivita committed:
fix: Extra newline for really long words

jlacivita committed:
feat: Play Intent (#151)

* First draft of Play Intent
* Fix document headers
* feat: Added Entity specs
* fix: Updating github.io publish script to handle index.md better
* Iterations from spec review
* fix: Iteration w/ Simon's feedback
* fix: Split play intent into play-entity and play-query
* fix: Adding assetId back to spec
* feat: Adding Play Intent schemas
* fix: Schema cleanup
* fix: adding validate:each to dist task
* fix: Remove enum form context.source
* fix: Fix links and formatting

semantic-release-bot committed:
chore(release): 0.18.0-next.5 [skip ci]

# [0.18.0-next.5](https://github.com/rdkcentral/firebolt-apis/compare/v0.18.0-next.4...v0.18.0-next.5) (2023-10-27)

### Features

* Play Intent ([#151](https://github.com/rdkcentral/firebolt-apis/issues/151)) ([d1ddf3f](https://github.com/rdkcentral/firebolt-apis/commit/d1ddf3fb3b1f758315686ad2f6dc57c2b270f33a))

jlacivita committed:
fix: Cleaned up Presentation spec

jlacivita committed:
Update presentation.md

jlacivita committed:
fix: Latest WG edits

satlead committed:
fix: Permission schema with Capability as required

satlead committed:
Merge pull request #198 from rdkcentral/capabilities-schema-fix

fix: Permission schema with Capability as required

semantic-release-bot committed:
chore(release): 0.18.0-next.6 [skip ci]

# [0.18.0-next.6](https://github.com/rdkcentral/firebolt-apis/compare/v0.18.0-next.5...v0.18.0-next.6) (2023-10-30)

### Bug Fixes

* Permission schema with Capability as required ([74d9de9](https://github.com/rdkcentral/firebolt-apis/commit/74d9de9b125baffb415e11ba3888c1c464cf5b12))

jlacivita committed:
fix: schemas version upgraded to 2.0.0-next.1 (#199)

semantic-release-bot committed:
chore(release): 0.18.0-next.7 [skip ci]

# [0.18.0-next.7](https://github.com/rdkcentral/firebolt-apis/compare/v0.18.0-next.6...v0.18.0-next.7) (2023-10-30)

### Bug Fixes

* schemas version upgraded to 2.0.0-next.1 ([#199](https://github.com/rdkcentral/firebolt-apis/issues/199)) ([b0fb8cf](https://github.com/rdkcentral/firebolt-apis/commit/b0fb8cfbda6bbef055000dc9911f344ab9ee8771))

jlacivita committed:
fix: Formatting

kpears201 commented on requirements/specifications/lifecycle/presentation.md:
```diff
@@ -0,0 +1,249 @@
+The loading screen **MUST** be displayed when the user attempts to launch the 
+app. 
+
+The loading screen **MUST** stay displayed until the app becomes active, or 
```
We should support hybrid. Even if they use an app provided loading screen, it will still take time to get the app provided loading screen downloaded and rendered. So maybe this should read the loading screen is displayed until the app becomes active or the app loading screen is ready, or launching is cancelled.

jlacivita committed:
fix: Updates form today's WG

jlacivita committed:
fix: Update schemas & 1.0.0 version! (#200)

fix: Update schemas

This is not really a breaking change, but we are finally shedding the 0.x major version in favor of 1.0.0!!!! 🎉

BREAKING CHANGE: This is not really a breaking change, but 1.0 release!!!

jlacivita committed:
fix: CI looking for legacy 0.x version

semantic-release-bot committed:
chore(release): 1.0.0-next.1 [skip ci]

# [1.0.0-next.1](https://github.com/rdkcentral/firebolt-apis/compare/v0.18.0-next.7...v1.0.0-next.1) (2023-11-01)

### Bug Fixes

* CI looking for legacy 0.x version ([923d02a](https://github.com/rdkcentral/firebolt-apis/commit/923d02ae96716a5272c5507e864f366012824642))
* Update schemas & 1.0.0 version! ([#200](https://github.com/rdkcentral/firebolt-apis/issues/200)) ([ea29f33](https://github.com/rdkcentral/firebolt-apis/commit/ea29f3323a22ffa36904bdc6cf6e76a31fa8cb76))

### BREAKING CHANGES

* This is not really a breaking change, but 1.0 release!!!

kpears201 committed:
fix: Provider error (#202)

* fix: Errors from providers not sent through transport

* fix: Errors from providers not sent through transport

semantic-release-bot committed:
chore(release): 1.0.0-next.2 [skip ci]

# [1.0.0-next.2](https://github.com/rdkcentral/firebolt-apis/compare/v1.0.0-next.1...v1.0.0-next.2) (2023-11-01)

### Bug Fixes

* Provider error ([#202](https://github.com/rdkcentral/firebolt-apis/issues/202)) ([4666a5e](https://github.com/rdkcentral/firebolt-apis/commit/4666a5ee56846f14dd1ba79dab82891505b5a2ec))

jlacivita committed:
Merge pull request #204 from rdkcentral/next

Release 1.0.0

jlacivita committed:
Update CHANGELOG.md

jlacivita committed:
Update CHANGELOG.md

semantic-release-bot committed:
chore(release): 1.0.0 [skip ci]

# [1.0.0](https://github.com/rdkcentral/firebolt-apis/compare/v0.17.1...v1.0.0) (2023-11-03)

### Bug Fixes

* Account.session params to be mandatory ([#196](https://github.com/rdkcentral/firebolt-apis/issues/196)) ([fc5c638](https://github.com/rdkcentral/firebolt-apis/commit/fc5c63886d9b4eb30b32c1edc75f0f6afe80a827))
* CI looking for legacy 0.x version ([923d02a](https://github.com/rdkcentral/firebolt-apis/commit/923d02ae96716a5272c5507e864f366012824642))
* Permission schema with Capability as required ([74d9de9](https://github.com/rdkcentral/firebolt-apis/commit/74d9de9b125baffb415e11ba3888c1c464cf5b12))
* Provider error ([#202](https://github.com/rdkcentral/firebolt-apis/issues/202)) ([4666a5e](https://github.com/rdkcentral/firebolt-apis/commit/4666a5ee56846f14dd1ba79dab82891505b5a2ec))
* schemas version upgraded to 2.0.0-next.1 ([#199](https://github.com/rdkcentral/firebolt-apis/issues/199)) ([b0fb8cf](https://github.com/rdkcentral/firebolt-apis/commit/b0fb8cfbda6bbef055000dc9911f344ab9ee8771))
* Update schemas & 1.0.0 version! ([#200](https://github.com/rdkcentral/firebolt-apis/issues/200)) ([ea29f33](https://github.com/rdkcentral/firebolt-apis/commit/ea29f3323a22ffa36904bdc6cf6e76a31fa8cb76))
* Updated Discovery.launch example with xrn:firebolt:application-type ([#187](https://github.com/rdkcentral/firebolt-apis/issues/187)) ([4dbbab3](https://github.com/rdkcentral/firebolt-apis/commit/4dbbab3d9fa68c0e5185ca72fd0170bae8a30139))
* Updated metrics:distributor capability to could ([#185](https://github.com/rdkcentral/firebolt-apis/issues/185)) ([9ca2206](https://github.com/rdkcentral/firebolt-apis/commit/9ca2206a2dca7149dcf93df3d303806d136785ed))

### Features

* Added optional parameter 'force' in usergrants.request ([#186](https://github.com/rdkcentral/firebolt-apis/issues/186)) ([31801ca](https://github.com/rdkcentral/firebolt-apis/commit/31801caec6bea0e8b295ea6a9ec54ca1d8e08d16))
* Play Intent ([#151](https://github.com/rdkcentral/firebolt-apis/issues/151)) ([d1ddf3f](https://github.com/rdkcentral/firebolt-apis/commit/d1ddf3fb3b1f758315686ad2f6dc57c2b270f33a))

### BREAKING CHANGES

* This is not really a breaking change, but 1.0 release!!!

kevinshahfws committed:
feature: Release NPM for PR (#207)

* feature: Release NPM for PR

kpears201 committed:
chore: Fix docs generated in prs (#212)

kpears201 committed:
chore: Improve documentation formatting (#211)

jlacivita committed:
Update NOTICE

jlacivita committed:
Update NOTICE

jlacivita committed:
fix: latest udpates

jlacivita committed:
feat: HDMI Low Latency Mode (#156)

* chore: Added OpenRPC Schemas
* fix: Move sources to CEC branch
* fix: Remove`Port` prefix for various APIs
* fix: Align onAutoLowLatencyModeCapableChanged event properties
* chore: Promoting to Proposed Specification
* fix: Leveraging global subscriber type
* feat: Adding EDID Version property
* fix: Added test
* fix: Add titles/declarations for various result schemas
* feat: Add support for HDMI Select
* fix: remove Hdmi Preview support

---------

Co-authored-by: Sathishkumar Deena Kirupakaran <Sathishkumar_deenakirupakaran@comcast.com>
Co-authored-by: Shah, Kevin <Kevin_Shah@comcast.com>

semantic-release-bot committed:
chore(release): 1.1.0-next.1 [skip ci]

# [1.1.0-next.1](https://github.com/rdkcentral/firebolt-apis/compare/v1.0.0...v1.1.0-next.1) (2024-01-16)

### Features

* HDMI Low Latency Mode ([#156](https://github.com/rdkcentral/firebolt-apis/issues/156)) ([06d3624](https://github.com/rdkcentral/firebolt-apis/commit/06d3624b69ee0529d4c1a1c78a15dbfdf54c8b16))

jlacivita committed:
fix: Add Music to Playable entities (#225)

* fix: Add Music to Playable entities
* fix: Adding Music to EntityIntent as well

semantic-release-bot committed:
chore(release): 1.1.0-next.2 [skip ci]

# [1.1.0-next.2](https://github.com/rdkcentral/firebolt-apis/compare/v1.1.0-next.1...v1.1.0-next.2) (2024-02-07)

### Bug Fixes

* Add Music to Playable entities ([#225](https://github.com/rdkcentral/firebolt-apis/issues/225)) ([22c9b71](https://github.com/rdkcentral/firebolt-apis/commit/22c9b71d3c0ee98d32585d1b365afabf8e64d6ed))

kevinshahfws committed:
fix: Modified account:uid to SHOULD (#224)

Co-authored-by: Jeremy LaCivita <jeremy.lacivita@comcast.com>

semantic-release-bot committed:
chore(release): 1.1.0-next.3 [skip ci]

# [1.1.0-next.3](https://github.com/rdkcentral/firebolt-apis/compare/v1.1.0-next.2...v1.1.0-next.3) (2024-02-07)

### Bug Fixes

* Modified account:uid to SHOULD ([#224](https://github.com/rdkcentral/firebolt-apis/issues/224)) ([70c8b24](https://github.com/rdkcentral/firebolt-apis/commit/70c8b24decfcbff2c32fb1b0d21290afc00a8432))

kschrief committed:
chore: OpenRPC 2.3.0 Update

kschrief committed:
Merge branch 'main' into next

kschrief committed:
Merge pull request #227 from rdkcentral/next

Next to Main - 1.1.0

kschrief committed:
Update CHANGELOG.md

kschrief committed:
Update CHANGELOG.md

semantic-release-bot committed:
chore(release): 1.1.0 [skip ci]

# [1.1.0](https://github.com/rdkcentral/firebolt-apis/compare/v1.0.0...v1.1.0) (2024-02-09)

### Bug Fixes

* Add Music to Playable entities ([#225](https://github.com/rdkcentral/firebolt-apis/issues/225)) ([22c9b71](https://github.com/rdkcentral/firebolt-apis/commit/22c9b71d3c0ee98d32585d1b365afabf8e64d6ed))
* Modified account:uid to SHOULD ([#224](https://github.com/rdkcentral/firebolt-apis/issues/224)) ([70c8b24](https://github.com/rdkcentral/firebolt-apis/commit/70c8b24decfcbff2c32fb1b0d21290afc00a8432))

### Features

* HDMI Low Latency Mode ([#156](https://github.com/rdkcentral/firebolt-apis/issues/156)) ([06d3624](https://github.com/rdkcentral/firebolt-apis/commit/06d3624b69ee0529d4c1a1c78a15dbfdf54c8b16))

jlacivita committed:
fix: Piotr's feedback/edits

kevinshahfws committed:
fix: Removed sdk as required field from Device.version (#231)

semantic-release-bot committed:
chore(release): 1.1.1-next.1 [skip ci]

## [1.1.1-next.1](https://github.com/rdkcentral/firebolt-apis/compare/v1.1.0...v1.1.1-next.1) (2024-02-28)

### Bug Fixes

* Removed sdk as required field from Device.version ([#231](https://github.com/rdkcentral/firebolt-apis/issues/231)) ([ef3de3e](https://github.com/rdkcentral/firebolt-apis/commit/ef3de3e68619c6f666dd1d751b0acc25f6438122))

jlacivita committed:
fix: Add PlaylistEntity to EntityIntent

semantic-release-bot committed:
chore(release): 1.1.1-next.2 [skip ci]

## [1.1.1-next.2](https://github.com/rdkcentral/firebolt-apis/compare/v1.1.1-next.1...v1.1.1-next.2) (2024-03-06)

### Bug Fixes

* Add PlaylistEntity to EntityIntent ([9769d6d](https://github.com/rdkcentral/firebolt-apis/commit/9769d6d7d43f1d61dfcda16ba4806175c5cf7658))

bpvstaty366 committed:
CoreSDK suite on MFOS - FCA standalone actions  (#234)

Core sanity suite execution from Mock-Firebolt on the PR

HaseenaSainul committed:
feat: Firebolt CPP SDKs

* CPPSDK: manage test added (#189)

* CPPSDK: standalone cpp test case added (#190)

* CPPTest: CXX compilor target added (#191)

* CPPSDK: test event sample test added (#192)

* CPPSDK: test event sample test added

* CPPSDK: more event test added

* CPPSDK: test updates based on interface name changes + disable rpc-only based methods/events (#205)

* CPPSDK: test case added for core (#209)

* CPPSDK: test case added for core

* Manage Test update based on provider integration

* Manage Test update based on provider integration

* Core test update to add more test case to link with provider sequence

* Core/Manage test app updates

* CPPSDK: templates added for metrics calls-metrics methods

* CPPSDK: test updates

* CPPSDK: Core test app updated with discovery.launch and watchNext

* CPPSDK: Core test cases added for discovery and capabilities

* CPPSDK: Core test cases updated based on openrpc generation support in features/cpp branch

* CPPSDK: install command added (#218)

* CoreTest: use has_value() check before accessing optional field (#222)

* use cpp header for strtok (#228)

* CPPSDK: use cpp header for strtok and cleanup (#230)

* CPP: update based on merge with next branch (#237)

* CPPSDK: updated Core test app to align with next branch merge changes (#238)

* CPPSDK: Async test case updates (#249)

* Pointed openrpc to cpp branch

* Pointed openrpc to cpp branch

* fix: Upgrade to latest firebolt-openrpc next build

---------

Co-authored-by: Santhosh Ramani <s.ramani@metrological.com>
Co-authored-by: Shah, Kevin <Kevin_Shah@comcast.com>
Co-authored-by: Jeremy LaCivita <jeremy.lacivita@comcast.com>

semantic-release-bot committed:
chore(release): 1.2.0-next.1 [skip ci]

# [1.2.0-next.1](https://github.com/rdkcentral/firebolt-apis/compare/v1.1.1-next.2...v1.2.0-next.1) (2024-03-27)

### Features

* Firebolt CPP SDKs ([5ea5d4e](https://github.com/rdkcentral/firebolt-apis/commit/5ea5d4e2f486a7bc3861955cd98323fc2c26cce1)), closes [#189](https://github.com/rdkcentral/firebolt-apis/issues/189) [#190](https://github.com/rdkcentral/firebolt-apis/issues/190) [#191](https://github.com/rdkcentral/firebolt-apis/issues/191) [#192](https://github.com/rdkcentral/firebolt-apis/issues/192) [#205](https://github.com/rdkcentral/firebolt-apis/issues/205) [#209](https://github.com/rdkcentral/firebolt-apis/issues/209) [#218](https://github.com/rdkcentral/firebolt-apis/issues/218) [#222](https://github.com/rdkcentral/firebolt-apis/issues/222) [#228](https://github.com/rdkcentral/firebolt-apis/issues/228) [#230](https://github.com/rdkcentral/firebolt-apis/issues/230) [#237](https://github.com/rdkcentral/firebolt-apis/issues/237) [#238](https://github.com/rdkcentral/firebolt-apis/issues/238) [#249](https://github.com/rdkcentral/firebolt-apis/issues/249)

kschrief committed:
feat: CI/CD docs (next-major) and merges (next-major/main) (#245)

* New workflow

* Fix: Not really

* Undo comments

* Undo comments

* Added merge workflows

* fix: Reversed version override flow

* Less logins on GH Actions

semantic-release-bot committed:
chore(release): 1.2.0-next.2 [skip ci]

# [1.2.0-next.2](https://github.com/rdkcentral/firebolt-apis/compare/v1.2.0-next.1...v1.2.0-next.2) (2024-04-08)

### Features

* CI/CD docs (next-major) and merges (next-major/main) ([#245](https://github.com/rdkcentral/firebolt-apis/issues/245)) ([49ab03d](https://github.com/rdkcentral/firebolt-apis/commit/49ab03db9986b27dafd279fcc6121abe447784dc))

kschrief committed:
chore: Update openRPC dependency (#262)

jlacivita committed:
Featuer/play intent (#272)

Standardizing Intent Message requirements

jlacivita committed:
feat: User Interest (#170)

In additional to traditional discovery APIs such as Watch History and
Watch Next, Firebolt provides a more abstract API that facilitates
impromptu content discovery connections between first-party Aggregated
Experiences and third-party Apps.

The User Interest Capability enables Apps to provide meta-data on
content that the user has expressed an interest in to Aggregated
Experience Apps that have been given access to use this Capability.

This allows for open ended design of Aggregated Experience App features
that present App-specific content to re-engage the user with the content
inside the originating App.

While the functionality and UX is left to the Aggregated Experience App,
typically designed by each Firebolt Distributor, the Firebolt API
enables events to register user interest and pass entity meta-data.

Which generally enables Aggregated Experiences to present that entity
meta-data in some way that leads to re-launching the original App at a
later point, using a `navigateTo` notification.

This is just one example of what an Aggregated Experience App might do
with the User Interest API.

Note that this API **SHOULD NOT** be used to implement Watch History or
Watch Next features. These concepts are much more fundamental to
Firebolt and have explicit APIs so that Firebolt Distributors can keep
track of which apps are using them separately.

semantic-release-bot committed:
chore(release): 1.2.0-next.3 [skip ci]

# [1.2.0-next.3](https://github.com/rdkcentral/firebolt-apis/compare/v1.2.0-next.2...v1.2.0-next.3) (2024-06-06)

### Features

* User Interest ([#170](https://github.com/rdkcentral/firebolt-apis/issues/170)) ([48a1094](https://github.com/rdkcentral/firebolt-apis/commit/48a1094aaab6418f09db662dbc81f090a34f32ed))

jlacivita committed:
feat: Command and Control Intents (#251)

* feat: Command and Control Intents

* Addition of the intents to control Firebolt compliant devices.

Control intents are for user intentions that will be needed regardless of whether there are any apps installed such as Power Intents, Volume Intents, Channel Intents, Media Control Intents

semantic-release-bot committed:
chore(release): 1.2.0-next.4 [skip ci]

# [1.2.0-next.4](https://github.com/rdkcentral/firebolt-apis/compare/v1.2.0-next.3...v1.2.0-next.4) (2024-06-06)

### Features

* Command and Control Intents ([#251](https://github.com/rdkcentral/firebolt-apis/issues/251)) ([c8f8dae](https://github.com/rdkcentral/firebolt-apis/commit/c8f8dae5a9a0f14a3815c04df5a55763823d4898))

jlacivita committed:
BREAKING CHANGE: First pass at Lifecycle 2.0 API schemas & tests

jlacivita committed:
Merge branch 'next' into major/lifecycle-improvements

jlacivita committed:
fix: Post-merge cleanup

jlacivita committed:
fix: Move HDMI, UserInterest over to new RPC paradigm

BREAKING CHANGE: previous breaking change commit had this in the wrong place, putting it here now.

jlacivita committed:
fix: Update CI to use new npm script name

jlacivita committed:
fix: Comment out sanity tests

jlacivita committed:
chore: Bumping version

jlacivita committed:
fix: Trigger major version build

jlacivita committed:
fix: Move transport interface back to string, not json

jlacivita committed:
fix: Remove unused example macros

jlacivita committed:
fix: Valid firebolt-openrpc reference

jlacivita committed:
fix: Change Sleeping to Hibernating

rdkcmf-jenkins commented:
## Blackduck scan failure details

**Summary**: 0 **violation**s, 0 files pending approval, 1 file pending identification.
* Protex Server Path: /home/blackduck/github/firebolt-apis/194/rdkcentral/firebolt-apis

* Commit: e74d1fe2d92e20687ec6fc118d6681b53f538f5f

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/7b38a987124665f2f9c5974bb900847a)

rdkcmf-jenkins commented:
## Blackduck scan failure details

**Summary**: 0 **violation**s, 0 files pending approval, 1 file pending identification.
* Protex Server Path: /home/blackduck/github/firebolt-apis/194/rdkcentral/firebolt-apis

* Commit: e74d1fe2d92e20687ec6fc118d6681b53f538f5f

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/5e5773d20d31c91b715a269dda8429cc)

rdkcmf-jenkins commented:
## **WARNING:** A Blackduck scan failure has been waived

A prior failure has been upvoted
* Upvote reason: OK

* Commit: e74d1fe2d92e20687ec6fc118d6681b53f538f5f

jlacivita committed:
fix: Update Hibernation based on last two discussions

jlacivita committed:
fix: Updates post-working group meeting
